src/DMXSerial2.cpp

```diff
@@ -638,69 +665,98 @@ void DMXSerialClass2::_processRDMMessage(byte CmdClass, uint16_t Parameter, bool
+      if (CmdClass == E120_GET_COMMAND) {
+        if (_rdm.packet.DataLength > 0) {
+          // Unexpected data
+          nackReason = E120_NR_FORMAT_ERROR;
+        } else if (_rdm.packet.SubDev != 0) {
+          // No sub-devices supported
+          nackReason = E120_NR_SUB_DEVICE_OUT_OF_RANGE;
+        } else {
+          // return the manufacturer label
+          _rdm.packet.DataLength = strlen(_initData->manufacturerLabel);
+          _rdm.packet.DataLength = min(_rdm.packet.DataLength, DMXSERIAL_MAX_RDM_STRING_LENGTH);
+          memcpy(_rdm.packet.Data, _initData->manufacturerLabel, _rdm.packet.DataLength);
+          handled = true;
+        }
+      } else if (CmdClass == E120_SET_COMMAND) {
+        // Unexpected set
+        nackReason = E120_NR_UNSUPPORTED_COMMAND_CLASS;
       }
 
-    } else if ((CmdClass == E120_GET_COMMAND) && (Parameter == SWAPINT(E120_DEVICE_MODEL_DESCRIPTION))) { // 0x0080
-      if (_rdm.packet.DataLength > 0) {
-        // Unexpected data
-        nackReason = E120_NR_FORMAT_ERROR;
-      } else if (_rdm.packet.SubDev != 0) {
-        // No sub-devices supported
-        nackReason = E120_NR_SUB_DEVICE_OUT_OF_RANGE;
-      } else {
-        // return the DEVICE MODEL DESCRIPTION
-        _rdm.packet.DataLength = strlen(_initData->deviceModel);
-        memcpy(_rdm.packet.Data, _initData->deviceModel, _rdm.packet.DataLength);
-        handled = true;
+    } else if (Parameter == SWAPINT(E120_DEVICE_MODEL_DESCRIPTION)) { // 0x0080
+      if (CmdClass == E120_GET_COMMAND) {
+        if (_rdm.packet.DataLength > 0) {
+          // Unexpected data
+          nackReason = E120_NR_FORMAT_ERROR;
+        } else if (_rdm.packet.SubDev != 0) {
+          // No sub-devices supported
+          nackReason = E120_NR_SUB_DEVICE_OUT_OF_RANGE;
+        } else {
+          // return the DEVICE MODEL DESCRIPTION
+          _rdm.packet.DataLength = strlen(_initData->deviceModel);
+          _rdm.packet.DataLength = min(_rdm.packet.DataLength, DMXSERIAL_MAX_RDM_STRING_LENGTH);
+          memcpy(_rdm.packet.Data, _initData->deviceModel, _rdm.packet.DataLength);
+          handled = true;
+        }
+      } else if (CmdClass == E120_SET_COMMAND) {
+        // Unexpected set
+        nackReason = E120_NR_UNSUPPORTED_COMMAND_CLASS;
       }
 
     } else if (Parameter == SWAPINT(E120_DEVICE_LABEL)) { // 0x0082
       if (CmdClass == E120_SET_COMMAND) {
         if (_rdm.packet.DataLength > DMXSERIAL_MAX_RDM_STRING_LENGTH) {
           // Oversized data
           nackReason = E120_NR_FORMAT_ERROR;
+        } else if (_rdm.packet.SubDev != 0) {
+          // No sub-devices supported
+          nackReason = E120_NR_SUB_DEVICE_OUT_OF_RANGE;
         } else {
           memcpy(deviceLabel, _rdm.packet.Data, _rdm.packet.DataLength);
-          deviceLabel[_rdm.packet.DataLength] = '\0';
+          deviceLabel[min(_rdm.packet.DataLength, DMXSERIAL_MAX_RDM_STRING_LENGTH)] = '\0';
```

Comment:
This check should be performed for the length of the memcpy too.

Reply:
This change is actually originally in #37 (I wanted to fix the issues OLA discovered independently of adding the personalities, as the former should be simpler to get in).

I think it effectively already is via the if statement above, but I went a bit belt and braces, perhaps it should be removed or restructured.